_layouts/homepage.html

```diff
@@ -114,4 +115,22 @@ <h1>Only 1 carousel is allowed per page.</h1>
 			{%- include homepage/carousel.html -%}
 		{%- endif -%}
 	{%- endif -%}
+
+	{%- if section.announcements -%}
+		{%- if has_announcements -%}
+			<section class="bp-section bg-newssection">
+				<div class="bp-container">
+					<div class="row">
+						<div class="col">
+							<h1>Each page should only have one announcements section</h1>
+						</div>
+					</div>
+				</div>
+			</section>
+		{%- else -%}
+			{%- include homepage/announcements.html -%}
+			{%- include functions/swap_container_class.html -%}
+			{%- assign has_announcements = true -%}
+		{%- endif -%}
```

Comment:
could you explain this for me? it's abit unclear what `swap_container_class` is doing (my guess is that it maintains the ordering of gray/non-gray?) 

also do take note that there's an ex**tra spacing after the last divider** even though `isLast` doesn't have the vertical margins. (see below ss)

could you also clarify why the assignment to `has_announcements` is made in the `else` block as it seems like it's not being used anywhere else?

![Screenshot 2023-09-14 at 5 36 25 PM](https://github.com/isomerpages/isomerpages-template/assets/44049504/832d1e27-310b-4123-9aca-94abe46cdbb3)

Reply:
> could you explain this for me? it's abit unclear what `swap_container_class` is doing (my guess is that it maintains the ordering of gray/non-gray?)

Yup, it is a special function used to change the background of the components, so that it alternates between a grey and white background across different homepage components.


> also do take note that there's an ex**tra spacing after the last divider** even though `forLoop.last` doesn't have the vertical margins. (see below ss)

Yup, the extra spacing is intended, which is the padding size of the entire announcements component. This is from the `p-16` class on line 4.

> could you also clarify why the assignment to `has_announcements` is made in the `else` block as it seems like it's not being used anywhere else?

This is to enforce that there is only one announcements component present in the homepage. Initially, `has_announcements` is set to `false` and the homepage will loop through all the components. When it first encounters an announcement block, it will display the announcements section and set `has_announcements` to `true`. If it encounters another announcement block, it will render the message "Each page should only have one announcements section".

This is something that the CMS will never encounter, because we will not allow the user to add more than one announcements block.